src/json_schema/mod.rs

```diff
@@ -1098,4 +1098,34 @@ mod tests {
         let regex = to_regex(&json_value, None);
         assert!(regex.is_ok(), "{:?}", regex);
     }
+
+    #[test]
+    fn test_uri_format() {
+        let schema = r#"{"title": "Foo", "type": "string", "format": "uri"}"#;
+        let json: Value = serde_json::from_str(schema).expect("Can't parse json");
+        let regex = to_regex(&json, None).expect("To regex failed");
+        let re = Regex::new(&regex).expect("Regex failed");
+
+        let valid_uris = vec![
+            "http://example.com",
+            "https://example.com/path?query=param#fragment",
+            "ftp://ftp.example.com/resource",
+            "urn:isbn:0451450523",
+        ];
+
+        let invalid_uris = vec![
+            "http:/example.com", // missing slash
+            "htp://example.com", // invalid scheme
+            "http://",           // missing host
+            "example.com",       // missing scheme
+        ];
+
+        for uri in valid_uris {
+            assert!(re.is_match(uri), "Valid URI failed: {}", uri);
+        }
+
+        for uri in invalid_uris {
+            assert!(!re.is_match(uri), "Invalid URI matched: {}", uri);
+        }
+    }
```

Comment:
Please, integrate these tests to be a part of the main `test_schema_matches_regex` test in  the end of the `String Type` section here:
https://github.com/dottxt-ai/outlines-core/blob/cbcb5e918a928b5d43714aede4cc86d63247cba1/src/json_schema/mod.rs#L353

Reply:
Done.
One question, same regex is in `parsing.rs` and in tests here, should we move it into some variable so that when we update it, can be done from one change itself ??

Reply:
Oh, I see, maybe it is supposed to be done via `FormatType`

Reply:
Yes exactly, and the you can refer to it in tests too (check how it's done for others)